### PR TITLE
allow getting/setting instruments on a context

### PR DESCRIPTION
**Pull request by jeffrand**
Signed-off-by: jeffrand <jeffr@seatgeek.com>

### CODE REVIEW

File: instrument.go [hunk 1]
```diff
@@ -253,10 +253,9 @@ func (i *InstrumentsConfig) ServeHTTP(rw http.ResponseWriter, r *http.Request, n
 	txn.AddAttribute("query", r.URL.RawQuery)
 
 	name := r.Method + " " + numberRegex.ReplaceAllString(r.URL.Path, "*")
-	instruments := i.WithTransaction(name, txn)
 
-	ctx := r.Context()
-	r = r.WithContext(context.WithValue(ctx, instrumentsCtx{}, instruments))
+	ctx, instruemnts := i.SetInstrumentsOnContext(r.Context(), txn, name)
```
Comment (sbeal) [inline instrument.go]: typo of `instruemnts`

File: instrument.go [hunk 2]
```diff
@@ -265,10 +264,21 @@ func (i *InstrumentsConfig) ServeHTTP(rw http.ResponseWriter, r *http.Request, n
 	timer.EndWithFields(FieldsList{"status": res.Status()})
 }
 
-// GetInstruments Returns the Intruments struct that is attached to a request
+func (i *InstrumentsConfig) SetInstrumentsOnContext(ctx context.Context, txn Transaction, name string) (context.Context, Instruments) {
```
Comment (lorenzo) [inline instrument.go]: Missing docs for the method

Comment (lorenzo) [reply instrument.go]: A good dock block would explain why this function was desirable

**Comment by lorenzo** [comment]
Are you doing this for testing purposes?

**Comment by jeffrand** [comment]
@lorenzo updated. I'm planning on using this in the catalog service, which doesn't use net.http to handle requests

**Comment by lorenzo** [comment]
LGTM

**Pull request merged**